src/main/java/com/blueapron/connect/protobuf/SchemaUtils.java

```diff
@@ -0,0 +1,48 @@
+package com.blueapron.connect.protobuf;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+public class SchemaUtils {
+  static SchemaAndValue flattenNestedStructs(SchemaAndValue connectData, String fieldDelimiter) {
+    SchemaBuilder flatSchemaBuilder = SchemaBuilder.struct();
```

Comment:
This is minor, but I would do `SchemaBuilder flatSchemaBuilder = SchemaBuilder.struct();` inside the `populateFlatSchema` and instead of passing it to that method, return it out of it. 
It just reads better and no reason to mutate an object outside of method when it can be returned. You could even call build() inside and return 'flatSchema` right away.

Reply:
The `populateFlatSchema` function is recursive, so it needs to receive schema builder from the outside, or we can break it down further... I can break it down, it'll probably be more readable. 👍